Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -1194,3 +1201,24 @@ void MainWindow::ShowMemcardManager()
 
   manager.exec();
 }
+
+void MainWindow::OnUpdateProgressDialog(QString title, int progress, int total)
+{
+  if (m_progress_dialog == nullptr)
+  {
+    m_progress_dialog = new QProgressDialog(m_render_widget);
+    m_progress_dialog->setLabelText(title);
```

Comment:
Could you move the text/maximum setting out of this conditional? This way, the caller can update the message/range without having to close and re-open the dialog.